File: card-battle-game-domain/src/main/java/com/dainws/games/cbg/domain/communication/Event.java
package com.dainws.games.cbg.domain.communication;

public class Event {

	private EventCode code;
	private EventDetails details;

	public Event(EventCode code) {
		this.code = code;
	}

	public Event(EventCode code, EventDetails details) {
		this.code = code;
		this.details = details;
	}

	public EventCode getCode() {
		return this.code;
	}

	public EventDetails getDetails() {
		return details;
	}
}

File: client/src/domain/events/Event.ts
import EventCode from '@/domain/events/EventCode';

abstract class Event<T> {
  private code: EventCode;
  private details: T;

  protected constructor(code: EventCode, details: T) {
    this.code = code;
    this.details = details;
  }

  public getCode() {
    return this.code;
  }

  public getDetails() {
    return this.details;
  }
}

export default Event;
